### PR TITLE
🪟 🎨 Show Source-defined `cursor` and `primary key` fields on new Stream Details panel

### LINKED ISSUE #20227: Stream Details panel - update predefined cursor and primary fields behavior
## Tell us about the problem you're trying to solve
At the moment, we don't show the selected cursor field in the new details panel(https://github.com/airbytehq/airbyte/pull/19219) if it was predefined and we can't change it.

## Describe the solution you’d like
After discussing possible UX solutions with @edmundito and @Upmitt we agreed:
- Fill in the Cursor column with disabled `<RadioButtons>'s`
- Select the cursor field which is predefined
- - Fill in the Primary Key column with disabled `<CheckBox>'s`
- Select the primary key field which is predefined

~~Since we don't have a "selected disabled" state in the current implementation of `<RadioButtons>', so need to update it first:
https://github.com/airbytehq/airbyte/issues/20089~~

Not relevant for `<RadioButton>`, the desired state is supported.
But still relevant for `<CheckBox>` component: https://github.com/airbytehq/airbyte/issues/20246 

## Describe the alternative you’ve considered or used
A clear and concise description of any alternative solutions or features you've considered or are using today.

### PR DESCRIPTION
**Pull request by dizel852**
## What
Resolves https://github.com/airbytehq/airbyte/issues/20227

Based on https://github.com/airbytehq/airbyte/pull/20335 (need to be merged first)

## How
Show Source-defined Cursor and PK fields on the new Stream Details panel, since we can't change them - they are with disabled state and tooltip.

_Notes:_ tooltip wasn't in the requirements, I've added it since it can be helpful for a user to explain why selected fields can't be changed by providing a `learn more` link.

Screenshot at Dec 10 05-22-20

### CODE REVIEW

**Comment by krishnaglick** [comment]
Can we change the mouse on disabled cursor fields `not-allowed`?

File: airbyte-webapp/src/components/ui/CheckBox/CheckBox.tsx [hunk 1]
```diff
@@ -1,49 +1,38 @@
...
-    checked={props.checked}
-    indeterminate={indeterminate}
-  >
-    <CheckBoxInput {...props} type="checkbox" />
-    {indeterminate ? <FontAwesomeIcon icon={faMinus} /> : props.checked && <FontAwesomeIcon icon={faCheck} />}
-  </CheckBoxContainer>
-);
+export const CheckBox: React.FC<CheckBoxProps> = ({ indeterminate, small, ...inputProps }) => {
+  const { checked, disabled, className } = inputProps;
+  const iconSize = small ? "sm" : "lg";
```
[the remark this answers is not in the document]
Comment (dizel852) [reply airbyte-webapp/src/components/ui/CheckBox/CheckBox.tsx:15]: This was changed in PR https://github.com/airbytehq/airbyte/pull/20335

https://github.com/airbytehq/airbyte/blob/f84d27f6f9c8603fafb7f0ab7c42c9f9573f570c/airbyte-webapp/src/components/ui/CheckBox/CheckBox.tsx

**Review by krishnaglick — APPROVED** [review]
LGTM

**Review by teallarson — COMMENTED** [review]
I like the Tooltip!  I wonder if it would be helpful to show for all elements in the column (ie: the disabled and off ones too, so a user knows why they can't interact with them)

I also cannot remember where we ended up on with whether to _always_ show these two columns, or only if the sync mode requires a cursor and/or primary key?  I see they are always present.

That said, I don't think either of those pieces are blocking this and code lgtm

**Pull request merged**